### PR TITLE
Start fresh training run if files for resuming training are missing

### PR DESCRIPTION
**Pull request by arnavgarg1**
If training fails during the first epoch, then `training_progress.json` will be missing and `training_checkpoints` will not have the `latest.ckpt` file. In this case, Ludwig throws errors. However, it might be better to indicate that model resumption failed, and that a fresh model training run is being created.

### CODE REVIEW

File: ludwig/trainers/trainer.py [hunk 1]
```diff
@@ -1236,6 +1240,25 @@ def set_steps_to_1_or_quit(self, signum, frame):
                 signal.signal(signal.SIGINT, self.original_sigint_handler)
             sys.exit(1)
 
+    def resume_files_exist(
+        self,
+        training_progress_tracker_path: str,
+        training_checkpoint_path: str,
+    ) -> bool:
```
Comment (justinxzhao) [inline ludwig/trainers/trainer.py:1247]: nit: Let's add a docstring here.

File: ludwig/trainers/trainer.py [hunk 2]
```diff
@@ -1236,6 +1240,25 @@ def set_steps_to_1_or_quit(self, signum, frame):
...
+        missing_files = []
+        # training_progress.json
+        if not path_exists(training_progress_tracker_path):
```
Comment (justinxzhao) [inline ludwig/trainers/trainer.py:1250]: Is there a reason why this doesn't need to be is_coordinator() guarded?

**Review by justinxzhao — APPROVED** [review]

File: tests/integration_tests/test_experiment.py [hunk 3]
```diff
@@ -658,6 +658,44 @@ def test_experiment_model_resume(tmpdir):
...
+        if missing_file == "training_progress.json":
+            os.remove(missing_file_path)
+        else:
+            shutil.rmtree(missing_file_path)
+    finally:
+        # Training should start a fresh model training run without any errors
+        experiment_cli(config, dataset=rel_path, model_resume_path=output_dir)
+
+    predict_cli(os.path.join(output_dir, "model"), dataset=rel_path)
+    shutil.rmtree(output_dir, ignore_errors=True)
```
Comment (justinxzhao) [inline tests/integration_tests/test_experiment.py:696]: nit: Remove? It's a tmpdir, is this automatically cleaned up?

File: ludwig/trainers/trainer.py (hunk 2 above)
Comment (arnavgarg1) [reply ludwig/trainers/trainer.py]: Fixed! It should be guarded

**Pull request merged**